featuretools/computational_backends/calculate_feature_matrix.py

```diff
@@ -194,42 +206,57 @@ def calculate_feature_matrix(features, entityset=None, cutoff_time=None, instanc
                                   num_per_chunk=num_per_chunk)
 
-    # if verbose, create progess bar
-    if verbose:
-        chunks = []
-        if num_per_chunk == "cutoff time":
-            for _, group in iterator:
-                chunks.append(group)
-        else:
-            for chunk in iterator:
-                chunks.append(chunk)
-
-        pbar_string = ("Elapsed: {elapsed} | Remaining: {remaining} | "
-                       "Progress: {l_bar}{bar}| "
-                       "Calculated: {n}/{total} chunks")
-        iterator = make_tqdm_iterator(iterable=chunks,
-                                      total=len(chunks),
-                                      bar_format=pbar_string)
+    chunks = []
+    if num_per_chunk == "cutoff time":
+        for _, group in iterator:
+            chunks.append(group)
+    else:
+        for chunk in iterator:
+            chunks.append(chunk)
+
     feature_matrix = []
-    backend = PandasBackend(entityset, features)
 
-    for chunk in iterator:
-        # if not using chunks, pull out the group dataframe
-        if isinstance(chunk, tuple):
-            chunk = chunk[1]
-        _feature_matrix = calculate_chunk(features, chunk, approximate,
-                                          entityset, training_window,
-                                          profile, verbose,
-                                          save_progress, backend,
-                                          no_unapproximated_aggs,
-                                          cutoff_df_time_var,
-                                          target_time, pass_columns)
-        feature_matrix.append(_feature_matrix)
-        # Do a manual garbage collection in case objects from calculate_chunk
-        # weren't collected automatically
-        gc.collect()
-    if verbose:
-        iterator.close()
+    if n_jobs != 1 or dask_kwargs is not None:
+        feature_matrix = parallel_calculate_chunks(chunks=chunks,
+                                                   features=features,
+                                                   approximate=approximate,
+                                                   training_window=training_window,
+                                                   verbose=verbose,
+                                                   save_progress=save_progress,
+                                                   entityset=entityset,
+                                                   n_jobs=n_jobs,
+                                                   no_unapproximated_aggs=no_unapproximated_aggs,
+                                                   cutoff_df_time_var=cutoff_df_time_var,
+                                                   target_time=target_time,
+                                                   pass_columns=pass_columns,
+                                                   dask_kwargs=dask_kwargs or {})
+    else:
+        backend = PandasBackend(entityset, features)
```

Comment:
for consistency, perhaps this else block should be in a function like `parallel_calculate_chunks` is above